master/buildbot/util/misc.py

```diff
@@ -51,3 +51,74 @@ def writeLocalFile(path, contents, mode=None):  # pragma: no cover
         if mode is not None:
             os.chmod(path, mode)
         file.write(contents)
+
+
+def parse_port_string(portstr):
+    list_args = []
+    dict_kwargs = {}
+    for el in portstr.split(':'):
+        elements_eq = el.split('=')
+        if len(elements_eq) == 2 and elements_eq[0] != '':
+            dict_kwargs[elements_eq[0]] = elements_eq[1]
+        elif len(elements_eq) <= 1:
+            list_args.append(elements_eq[0])
+        else:
+            raise Exception('unsupported porstr {}'.format(portstr))
+    return (list_args, dict_kwargs)
+
+
+def get_host_port(input):
+    # return a tuple (host, port)
```

Comment:
why something custom?
twisted already has the concept of connection string. shouldn't we ust use that?

Reply:
In msgpack protocol worker needs host and port for establishing connection. This function parses both from Twisted connection string so that we can build connection string usable for autobahn.

With @p12tic we discussed that it may make sense to change API so that user specifies host, port and other information explicitly. Hence, this function would not be needed.